# METAGENOMIC ANALYSIS WITH MGRAST

## Introduction

MG-RAST is a free open source in the web where we can upload our data and it has different tools that will analyze them. It serves as a repository for datasets.

## Using MG-RAST

In this section I will describe the analysis elaborated by MG-RAST for the following a shotgun readings of the sample called "2L"that is part of the project: "Common Bean Rhizosphere Metatranscriptome" lead by Lucas William Mendes.

https://www.mg-rast.org/mgmain.html?mgpage=project&project=mgp20659

The first graphic we encounter is realted to the quality of the sequences, in this case 33.68% of the sequences does not meet the quality requirements, and 58.62% were used to predict different functions.

<p align="center"><img src="/IMAGES/nt2/mgrastqc.JPG"></p>

Taking that 58.62% Mg-RAST breaks it in the different traits that can be predicted, for example about 57.77% are unknow proteins and 38.65% are related to known proteins. We can see this distribution on figure 2.

<p align="center"><img src="/IMAGES/nt2/mgrastfeatures.JPG"></p>

Figure 3 shows an estimation of the sequencing error which as mentioned in the manual is independent of the sequencing method, currently is only available for metagnomic sequencing but no for amplicons.I our sample it found a 4.189% of error which for me sounds good, although I am not sure what is a aceptable range.

<p align="center"><img src="/IMAGES/nt2/mgrasterror.JPG"></p>

Another figure is the K-mer profile. For the figure the kmer coverage is equivalent to number of observations of each kmer and  is ploteed against the sequences size.

<p align="center"><img src="/IMAGES/nt2/mgrastkmerprofile.JPG"></p>

The nucleotide distribution chart shows the fraction of each nucelotide (A,G,T,C or N) at each position along the sequences. The manual says that for shotgun the distribution is expected to be closely in equal proportions.

<p align="center"><img src="/IMAGES/nt2/mgrastnucleotide.JPG"></p>

MG-rast uses diferent databases to aligns sequences to determine functionalities. The folowing figures shows the different profiles obtained using diferent databases.
<table>
  <td><p align="center"><img src="/IMAGES/nt2/mgrastko.JPG"></p></td> 
  <td><p align="center"><img src="/IMAGES/nt2/mgrastcog.JPG"></p></td>
 <tr>
   <td> Function profile using KO (Kyoto Encyclopedia of Genes and Genomes) database</td>
   <td> Function profile using COG (Clusters of Orthologous Groups) database</td>
  </tr>
  </table>


Finally MG-RAST uses the sequences that **Best** hit to predict the taxonomic and functional distribution of the sample, the algorithm searches the nonredundant M5nr and M5rna databases in which each sequence is unique. For our sample at the phylum level is mainly composed of Antinobacteria followed by Proteobacteria and Chordata.

<p align="center"><img src="/IMAGES/nt2/mgrastphylum.JPG"></p>

