# METAGENOMIC ANALYSIS WITH HUMANN2

## Introduction
hUMANn2 allows the quantification and profiling of genes families and pathways that comes from both chracterized and unchracterized microbiome members. Humann2 takes a whole metagenomic information then interprets this information and  translated these sequencing reads to associated them with metabolic functions.

The following Scheme is the flow description of how Humann2 proceeds (Franzosa et al., 2018)

ChocoPhlAn database is where the information for functional profiling is stored
Uniref database is the information used to annotate gene families and pathways


## Running the code

First we need to create a directory for the results, I will create mine into the SRR100123 directory that has the metagenomic information to be analyzed.
 - Looking into SRR100123
        
        bash-4.2$ cd SRR100123
        bash-4.2$ ls
        fastqcStuff
        bash-4.2$ cd fastqcStuff/
        bash-4.2$ ls
        SRR100123.fastq  SRR100123_fastqc.html  SRR100123_fastqc.zip
        
  - Back to the main directory, and create Humann2 output directory
        
        bash-4.2$ cd ..
        bash-4.2$ pwd
        /home/CAM/mcb5896/usr11/SRR100123
        bash-4.2$ mkdir Humann2_output
        bash-4.2$ ls
        fastqcStuff Humann2_output
    
As second step we load the necessary modules:

        bash-4.2$ module load humann2/0.11.1
        (1.70) 
        bash-4.2$ module load bowtie2/2.3.3.1
        (1.70)

      
Finally, we run the following code:

    bash-4.2$ humann2 --input /home/CAM/mcb5896/usr11/SRR100123/fastqcStuff/SRR100123.fastq --output /home/CAM/mcb5896/usr11/SRR100123/Humann2_output --nucleotide-database /home/CAM/mcb5896/classFiles/humann2_database_downloads/chocophlan --protein-database /home/CAM/mcb5896/classFiles/humann2_database_downloads/uniref/ --metaphlan-options "--mpa_pkl /home/CAM/mcb5896/classFiles/db_v20/ --bowtie2db /home/CAM/mcb5896/classFiles/db_v20/"

   
Breaking the code apart:


**-- input $filesample**  is a filtered shotgun sequencing metagenome file (fastq, fastq.gz, fasta, or fasta.gz format) that we need to analyze, it can also be mapping results file (sam, bam or blastm8 format) or gene table file (tsv or biom format).
 
**-- output $$Output** directory is the directory where we want the results
 
 the following three entries allows to assign the location for the databases if they are saved in a different directory than the one we are running Humann2.
 
**-- nucleotide-database $location** for ChocoPhlAn database

**-- protein-database $location** for RefSeq
 
**-- metaphlan-options**
 
 
The following is what the screen will print once running the code:
 
 
        Output files will be written to: /home/CAM/mcb5896/usr11/SRR100123/Humann2_output
        Removing spaces from identifiers in input file ...

        Running metaphlan2.py ........
        
        <div class="text-green">
        Found g__Prevotella.s__Prevotella_copri : 53.59% of mapped reads
        Found g__Bacteroides.s__Bacteroides_stercoris : 13.35% of mapped reads
        Found g__Collinsella.s__Collinsella_unclassified : 9.25% of mapped reads
        Found g__Bacteroides.s__Bacteroides_coprocola : 9.20% of mapped reads
        Found g__Bacteroides.s__Bacteroides_coprophilus : 5.87% of mapped reads
        Found g__Bacteroides.s__Bacteroides_vulgatus : 4.78% of mapped reads
        Found g__Alistipes.s__Alistipes_putredinis : 3.39% of mapped reads
        Found g__Megamonas.s__Megamonas_unclassified : 0.57% of mapped reads
        </div>

        Total species selected from prescreen: 8

        Selected species explain 100.00% of predicted community composition


        Creating custom ChocoPhlAn database ........


        Running bowtie2-build ........


        Running bowtie2 ........

        Total bugs from nucleotide alignment: 2
        g__Bacteroides.s__Bacteroides_vulgatus: 2126 hits
        g__Bacteroides.s__Bacteroides_stercoris: 1240 hits

        Total gene families from nucleotide alignment: 1484

        Unaligned reads after nucleotide alignment: 97.7927000407 %


        Running diamond ........


        Aligning to reference database: uniref90_demo_prots.dmnd

        Total bugs after translated alignment: 3
        g__Bacteroides.s__Bacteroides_vulgatus: 2126 hits
        unclassified: 113 hits
        g__Bacteroides.s__Bacteroides_stercoris: 1240 hits

        Total gene families after translated alignment: 1537

        Unaligned reads after translated alignment: 97.7389274332 %


        Computing gene families ...

        Computing pathways abundance and coverage ...

        Output files created:
        /home/CAM/mcb5896/usr11/SRR100123/Humann2_output/SRR100123_genefamilies.tsv
        /home/CAM/mcb5896/usr11/SRR100123/Humann2_output/SRR100123_pathabundance.tsv
        /home/CAM/mcb5896/usr11/SRR100123/Humann2_output/SRR100123_pathcoverage.tsv

## RESULTS

Know we have a view into the results obtaining:

**SRR100123_genefamilies.tsv**

    #Gene Family						                            SRR100123_Abundance-RPKs

    UNMAPPED					                                          	149046.0000000000

    UniRef90_A6KY67	1000.0000000000
    UniRef90_A6KY67|g__Bacteroides.s__Bacteroides_vulgatus	   1000.0000000000

    UniRef90_unknown705.3261217587
    UniRef90_unknown|g__Bacteroides.s__Bacteroides_vulgatus	   461.6156181611
    UniRef90_unknown|g__Bacteroides.s__Bacteroides_stercoris   243.7105035976

    UniRef90_R6AT12	333.3333333333
    UniRef90_R6AT12|g__Bacteroides.s__Bacteroides_stercoris	   333.3333333333

    UniRef90_Q650D4	118.5897435897
    UniRef90_Q650D4|g__Bacteroides.s__Bacteroides_stercoris	   118.5897435897

    UniRef90_R6AZX2	100.0000000000
    UniRef90_R6AZX2|g__Bacteroides.s__Bacteroides_stercoris	   100.0000000000

    UniRef90_E6SQI7	62.5000000000
    UniRef90_E6SQI7|g__Bacteroides.s__Bacteroides_stercoris	    62.5000000000

    UniRef90_R6AAG0	58.8235294118
    UniRef90_R6AAG0|g__Bacteroides.s__Bacteroides_stercoris	    58.8235294118

    UniRef90_R7NVM4	52.6315789474
    UniRef90_R7NVM4|g__Bacteroides.s__Bacteroides_vulgatus	    52.6315789474

    UniRef90_D4V4W8	50.0000000000
    UniRef90_D4V4W8|g__Bacteroides.s__Bacteroides_vulgatus	    50.0000000000

    UniRef90_A6L6M5	41.6666666667
    UniRef90_A6L6M5|g__Bacteroides.s__Bacteroides_vulgatus	    41.6666666667

    UniRef90_A6L0E1	37.0370370370
    UniRef90_A6L0E1|g__Bacteroides.s__Bacteroides_vulgatus	    37.0370370370

    UniRef90_D4V539	37.0370370370
    UniRef90_D4V539|g__Bacteroides.s__Bacteroides_vulgatus	    37.0370370370

    UniRef90_P70882	34.9123619909
    UniRef90_P70882|g__Bacteroides.s__Bacteroides_stercoris	    18.5361804588
    UniRef90_P70882|g__Bacteroides.s__Bacteroides_vulgatus	    16.3761815322

**SRR100123_pathabundance.tsv**

    # Pathway	SRR100123_Abundance
    UNMAPPED	29357.1078825168
    UNINTEGRATED	1308.1220941037
    UNINTEGRATED|unclassified	21.0805720326
    PWY-5695: urate biosynthesis/inosine 5'-phosphate degradation	2.4099112316
    PWY-5695: urate biosynthesis/inosine 5'-phosphate degradation|unclassified	2.1920072521

**SRR100123_pathcoverage.tsv**

    # Pathway	SRR100123_Coverage
    UNMAPPED	1.0000000000
    UNINTEGRATED	1.0000000000
    UNINTEGRATED|unclassified	1.0000000000
    PWY-5695: urate biosynthesis/inosine 5'-phosphate degradation	0.5578499913
    PWY-5695: urate biosynthesis/inosine 5'-phosphate degradation|unclassified	0.4615383106
    
    
## References

* Franzosa EA*, McIver LJ*, Rahnavard G, Thompson LR, Schirmer M, Weingart G, Schwarzberg Lipson K, Knight R, Caporaso JG, Segata N, Huttenhower C. Species-level functional profiling of metagenomes and metatranscriptomes. Nat Methods 15: 962-968 (2018).

