# METAGENOMIC READS QUALITY ASSESMENT

**Objective:**

1. Asses the quality of the reads obtained after sequencing 

## FASTQC

The first step after getting the readings is to evaluate the quality of the data. 
In the cluster we need a new directory (I named mine as Metagenomics)


        bash-4.2$ pwd
        /home/CAM/mcb5896/usr11
        bash-4.2$ ls
        
        chipmunk_fastq  mothur.1567784410.logfile  mothur.1571401653.logfile
        fastqcStuff     mothur.1567875472.logfile  mothur.1571417701.logfile
        Miseq_batch     mothur.1570745423.logfile  SRR100123.fastq
        MiSeq_SOP       mothur.1571018318.logfile  transferThisText.txt
        
        bash-4.2$ mkdir Metagenomics
        bash-4.2$ cd Metagenomics/

  And moved into it the reads that I will be analyzing
  
       
        bash-4.2$ cp /home/CAM/mcb5896/classFiles/HMP_1.fastq.gz .
        bash-4.2$ ls
        HMP_1.fastq.gz
        
Notice that the file is compressed and then we need to "unziped" it.


        bash-4.2$ cd Metagenomics/
        bash-4.2$ ls
        HMP_1.fastq.gz
        bash-4.2$ gunzip HMP_1.fastq.gz
        bash-4.2$ ls
        HMP_1.fastq

And know we load and run the fastqc module Version 0.11.7 (you will have to installed before if it is not present in the cluster)

        bash-4.2$ module load fastqc/0.11.7
        bash-4.2$ fastqc /home/CAM/mcb5896/usr11/Metagenomics/HMP_1.fastq
      
        Started analysis of HMP_1.fastq
        Approx 10% complete for HMP_1.fastq
        Approx 25% complete for HMP_1.fastq
        Approx 35% complete for HMP_1.fastq
        Approx 50% complete for HMP_1.fastq
        Approx 60% complete for HMP_1.fastq
        Approx 75% complete for HMP_1.fastq
        Approx 90% complete for HMP_1.fastq
        Analysis complete for HMP_1.fastq

As a result we have the following files:

        HMP_1.fastq  HMP_1_fastqc.html  HMP_1_fastqc.zip

Then the HMP_1_fastqc.html file has the information about the quality of our sequenced read and is called the FASTQC REPORT, we need to transfer it into our computers and it can be analyzed using any Web Brownser.

The result consist of a series of parameters that describe the quality of the run.
At the top of the document we found a table with a summary of Basic Statistic as follows:

<p align="center"><img src="/IMAGES/nt2/nt2_1.JPG"></p>

From these table we get that our reads have a content of 45% GC, if we were looking at a specific strain this data might be useful to known, from example is know that E. coli is high in GC content.

Another Interesting metric is the "Per base Sequence Quality"

<p align="center"><img src="/IMAGES/nt2/nt2_2.JPG"></p>

This barplot shows the distribution of base quality along the sequences. As can be seen the bases have a good quality score along most bases, dropping below 20 at the last 4 bases. This is a good quality score distribution.

Finally, a third graphic that gives a similar information is the "Per Sequence Quality Score" graph. This one shows a peak to the right near a quality score around 36, meaning that most of the reads have good quality scores,  if the graphic shows peaks in 20 or less quality scores should be something to evaluate in the process of trimming.

        
<p align="center"><img src="/IMAGES/nt2/nt2_3.JPG"></p>


For more information about Quality Scores and analysis of this graphic  we can check this video

(https://www.youtube.com/watch?v=GnWSXwQeJ_U)  from Andrew Oler, pHD.


Back to home




