# METAGENOMIC ANALYSIS WITH RAY

## INTRODUCTION

Ray is an assembler that use some reads and called them "seeds" based on these, it starts building contigs, different that de Brujin graph, where the contigs can take different paths at some points instead of "drawing all possible paths" this asembler will stop where a no clear path is encounter.

One of principal chracteristics that we need to know is that the algoritm uses K-mers (strings of nucleotides of a "k" lenght)

## RUNNING THE CODE using BATCH SCRIPT

Many times is better to run a specific code using a **batch script** rather than running the codes one by one in the cluster. This is specially important if the processing time is long which requires that we stay login while the code is running, otherwise it will stop.
Batch Scripts in the other hand run the code even when we log off the cluster.

The Following is the content of a Batch Script to run RAY, the first 12 lines are instructions for the cluster about the job, 
It describes the name of the job, computer power required to run it, an email where to receive notifications about the job and output and error files that describe the job outcome. The last part is the actual code syaing to load the module, get into a directory called raystuff and finally to run Ray. We can create and save this instructions as a text file. I created mine in Notepad and called it Ray_batch.sh

            #!/bin/bash
            #SBATCH --job-name=RayTest33
            #SBATCH -N 1
            #SBATCH -n 1
            #SBATCH -c 1
            #SBATCH -p mcbstudent
            #SBATCH --qos=mcbstudent
            #SBATCH --mail-type=ALL
            #SBATCH --mem=5G
            #SBATCH --mail-user=validemail@uconn.edu
            #SBATCH -o RayTest33_%j.out
            #SBATCH -e RayTest33_%j.err

            module load Ray/2.3.1
            mkdir raystuff
            cd rayStuff
            Ray -k 33 -s /home/CAM/mcb5896/classFiles/SRR100123.fastq -o SRR_test33

To actually run these command, we have to get into the cluster and go into the desired directory from which we want to run the code, here we have to SCP (secure copy) the Batch Script into these directory to actually run the code 

            bash-4.2$ pwd
            /home/CAM/mcb5896/usr11
            bash-4.2$ ls
            chipmunk_fastq  Metagenomics  MiSeq_SOP  PS_results_HMP1  RDP        SRR100123.fastq
            humann2Stuff    Miseq_batch   Project    Ray_batch.sh     SRR100123  transferThisText.txt
            bash-4.2$ sbatch Ray_batch.sh

You will receive a job number assigned and a notification in the e-mailbox when the code start running (it will go to the queue first) and when it sucessfully finish or if it failed.

## Picking the best K-mer

From the outputs we check the OutputNumbers.txt file that has information about the assembly at the K-mer pick. In class we run it with differente K-mer numbers, the following images shows the results per each k-mer:

<p align="center"><img src="/IMAGES/nt2/k-merchart.jpg"></p>

To pick the best K mer we have to check the different parameters

**Number** : How many contigs are found based on a k-mer. In the figure for K-mers of 100 nucleotides we have more contigs, but if it is 500 nt the number of contigs are lower. It can be a good change to have less contigs but made out of more nucleotides.

**Total length**: How many bases have been assembled into the contigs. More = better. In general for this dataset the lower the k-mer the more bases have been used for the assemblies. Interestingly for  k-mer size =21 made of 500nt the total lenght cover is greather than k-mer 19 made of the same time 500 nt. 

**Average**: This is the average size of the contigs in the assembly. Higher = better

**N50**:"N50 can be described as a weighted median statistic such that 50% of the entire assembly is contained in contigs or scaffolds equal to or larger than this value." wikipedia

**Median**: The median length of the contigs in the assembly 

**Largest** Length of the longest assembly

## Predicting the best K-mer with Kmergenie

We can also predict the best K-mer running Kmergenie 

A) installing Kmergenie
            getting the installer/unzipping/ and run test (make)
            
            cp /home/CAM/mcb5896/classFiles/kmergenie-1.7051.tar.gz . 
            gunzip -c kmergenie-1.7051.tar.gz | tar xopf â€“
            cd kmergenie-1.7051
            make
            
   The screen will print the following:
   
            Testing presence of ntCard....
            OK
            Testing presence of Rscript....
            R scripting front-end version 3.5.1 (2018-07-02)
            OK
            Testing basic Rscript functionality....
            Rscript --no-init-file -e 'rnorm(1)'
            [1] "rnorm(1)"
            OK
            Testing a simple KmerGenie example....

            OK
            Test successful. KmerGenie is ready, type `./kmergenie`.
    
  Know we can run the predictor
             
             ./kmergenie /home/CAM/mcb5896/classFiles/SRR100123.fastq
             
  The predicted best K-mer is print here:
  
            running histogram estimation
            Setting maximum kmer length to: 531 bp
            computing histograms (from k=21 to k=121): 21 61 71 121 91 111 101 81 51 41 31 
            ntCard wall-clock time over all k values: 169 seconds 
            fitting model to histograms to estimate best k
            estimation of the best k so far: 21
            refining estimation around [15; 27], with a step of 2
            running histogram estimation
            Setting maximum kmer length to: 531 bp
            computing histograms (from k=17 to k=27): 27 23 21 25 17 19 
            ntCard wall-clock time over all k values: 54 seconds 
            fitting model to histograms to estimate best k
            table of predicted num. of genomic k-mers: histograms.dat
            recommended coverage cut-off for best k: 15
            best k: 17
            
   An histogram.html result can be seen in the web brownser
   
<p align="center"><img src="/IMAGES/nt2/kmerpredictor.jpg"></p>


