# METAGENOMIC ANALYSIS WITH HUMANN2

## Introduction
hUMANn2 allows the quantification and profiling of genes families and pathways that comes from both chracterized and unchracterized microbiome members. Humann2 takes a whole metagenomic information then interprets this information and  translated these sequencing reads to associated them with metabolic functions.

The following Scheme is the flow description of how Humann2 proceeds (Franzosa et al., 2018)

ChocoPhlAn database is where the information for functional profiling is stored
Uniref database is the information used to annotate gene families and pathways

## Running the code
Loading the necessary modules:

        module load  humann2/0.11.1
        module load bowtie2/2.3.3.1

First we need to create a directory for the results

Then we run the following code

    humann2 --input /home/CAM/mcb5896/classFiles/demo.fastq --output output_demo --nucleotide-database /home/CAM/mcb5896/classFiles/humann2_database_downloads/chocophlan --protein-database /home/CAM/mcb5896/classFiles/humann2_database_downloads/uniref/ --metaphlan-options "--mpa_pkl /home/CAM/mcb5896/classFiles/db_v20/ --bowtie2db /home/CAM/mcb5896/classFiles/db_v20/"
    
Breaking the code apart
 -- input $filesample  is a filtered shotgun sequencing metagenome file (fastq, fastq.gz, fasta, or fasta.gz format) that we need to analyze, it can also be mapping results file (sam, bam or blastm8 format) or gene table file (tsv or biom format).
 
 -- output $$Output directory is the directory where we want the results
 
 the following three entries allows to assign the location for the databases if they are saved in a different directory than the one we are running Humann2.
 
 -- nucleotide-database $location for ChocoPhlAn database
 -- protein-database $location for RefSeq
 
 -- metaphlan-options
 
## References

* Franzosa EA*, McIver LJ*, Rahnavard G, Thompson LR, Schirmer M, Weingart G, Schwarzberg Lipson K, Knight R, Caporaso JG, Segata N, Huttenhower C. Species-level functional profiling of metagenomes and metatranscriptomes. Nat Methods 15: 962-968 (2018).

