## OBJECTIVE

## Checking the sequencing

Every experiment needs a control, in this case to know if the sequencing was good, along whit our samples we need to sequence a "mock community" by determining how well the sequencing of this community was performed we can say also how well our samples sequencing was. There is no allways the case where a mock community is run. Since, in our dataset we do not have a mock community, I am going to show the steps but not actuall results will came out of them.

**1. Calling the mock community **

to obtaing the  mock community that is part of our data we use:

    mothur> get.groups(count=stability.trim.contigs.good.unique.good.filter.unique.precluster.denovo.vsearch.pick.pick.count_table, fasta=stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta, groups=Mock)

NOTE: for our case of studio the dataset does not contain a mock community, thus this step will generate an empty file

OUTPUT:
<ul> 
  <li>stability.trim.contigs.good.unique.good.filter.unique.precluster.denovo.vsearch.pick.pick.pick.count_table</li>
  <li>stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.pick.fasta</li>
</ul>
  
  the count table tell us how many unique sequences and how many reads we have
  
**2. Comparing the unique sequences to what we know of the mock community

mothur > seq.error(fasta=stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.pick.fasta, count=stability.trim.contigs.good.unique.good.filter.unique.precluster.denovo.vsearch.pick.pick.pick.count_table, reference=HMP_MOCK.v35.fasta, aligned=F)
