## OBJECTIVE

## Checking the sequencing

Every experiment needs a control, in this case to know if the sequencing was good, along whit our samples we need to sequence a "mock community" by determining how well the sequencing of this community was performed we can say also how well our samples sequencing was. There is no allways the case where a mock community is run. Since, in our dataset we do not have a mock community, I am going to show the steps but not actuall results will came out of them.

**1. Calling the mock community**

To obtaing the  mock community that is part of our data we use:

    mothur> get.groups(count=stability.trim.contigs.good.unique.good.filter.unique.precluster.denovo.vsearch.pick.pick.count_table, fasta=stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta, groups=Mock)

NOTE: for our case of studio the dataset does not contain a mock community, thus this step will generate an empty file (figure 1)
<img src=nt4_1.jpg, figure 1>

Output:
    <ul> 
     <li>stability.trim.contigs.good.unique.good.filter.unique.precluster.denovo.vsearch.pick.pick.pick.count_table</li>
     <li>stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.pick.fasta</li>
    </ul>
  
The *"XXX.count_table"* file tell us how many unique sequences and reads we have.
  
**2. Comparing the unique sequences to what we know of the mock community**

In this step we compare the counts we have in the *".count_table"* with the information we have of the mock community. For instance, when preparing this library we knew how many different strains or species we put into it and at what proportion. Thus in the hypothetic case that our sequencing did not have errors, the count_file should show these known proportions. The HMP_MOCK.V35.FASTA contains the expected data of our community use for comparison.

    mothur > seq.error(fasta=stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.pick.fasta, count=stability.trim.contigs.good.unique.good.filter.unique.precluster.denovo.vsearch.pick.pick.pick.count_table, reference=HMP_MOCK.v35.fasta, aligned=F)

Output:
    <ul>
    <li> stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.pick.error.summary </li>
    <li> stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.pick.error.seq </li>
    <li> stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.pick.error.chimera </li>
    <li> stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.pick.error.seq.forward </li>
    <li> stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.pick.error.seq.reverse </li>
    <li> stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.pick.error.count </li>
    <li> stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.pick.error.matrix </li>
    <li> stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.pick.error.ref </li>
    </ul>

The summary file contains the imformation that is more important with the results of the errors.

**3. Mock Community OTUS information**

To check how many OTUS were detected in our sequenced Mock community we use a set of 4 commands

    mothur> dist.seqs(fasta=stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.pick.fasta, cutoff=0.03)
    
Output 1:

   <ul><li>stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.pick.dist</li></ul>
 

    mothur> cluster(column=stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.pick.dist, count=stability.trim.contigs.good.unique.good.filter.unique.precluster.denovo.vsearch.pick.pick.pick.count_table)
    
Output 2:     
    <ul>
    <li>stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.pick.opti_mcc.list</li>
    <li>stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.pick.opti_mcc.steps</li>
    <li>stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.pick.opti_mcc.sensspec</li>
    </ul>

    mothur> make.shared(list=stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.pick.opti_mcc.list, count=stability.trim.contigs.good.unique.good.filter.unique.precluster.denovo.vsearch.pick.pick.pick.count_table, label=0.03)

Output 3: 
    
   <ul>
   <li>stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.pick.opti_mcc.shared</li>
   </ul>
    
    mothur> rarefaction.single( shared=stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.pick.opti_mcc.shared
    
Output 4:
    
   <ul>
   <li>stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.pick.opti_mcc.groups.rarefaction</li>
   </ul>
    
To view the results we use:
    
    mothur > system(cat stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.pick.opti_mcc.groups.rarefaction )

The following images shows the results of a mock community and their sequencing results.

<img src=nt4_3>
<imag src=nt4_4>
    
the last row shows that from a total of 4048 sequences it was detected 35O OTUS, the physical community was prepared mixing 20 different species of bacteria. According to Shloss this a good result.

Once stimated the error of the sequencing and data handle, we dont need these sequences. Thus, to removed them from our data set we use:

    mothur > remove.groups(count=stability.trim.contigs.good.unique.good.filter.unique.precluster.denovo.vsearch.pick.pick.count_table, fasta=stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta, taxonomy=stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pds.wang.pick.taxonomy, groups=Mock)

<img src=nt4_5.jpg>

Output:

<ul>
<li> stability.trim.contigs.good.unique.good.filter.unique.precluster.denovo.vsearch.pick.pick.pick.count_table </li>
<li>stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.pick.fasta</li>
<li>stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pds.wang.pick.pick.taxonomy</li>
</ul>

## OTUS picking

