# ERROR STIMATION, OTUS PICKING, AND PHYLOGENETIC TREE 

**Recapitulating:** Last section we left after assigning taxonomies to our unique sequences.

### Objective:
<ul>
    <li>Stimate the error rate of the sequencing based on information retrieve from a mock community.</li>
    <li>Group our sequences into OTUS</li>
    <li>Create a phylogenetic tree from OTUS for future data analysis based on distance metrics.</li>
</ul>

## Checking the sequencing

Every experiment needs a control, in this case to know if the sequencing was good, along whit our samples we need to sequence a "mock community" by determining how well the sequencing of this community was performed we can say also how well our samples sequencing was. Nevertheless often a mock community is not sequenced. Since, in our dataset we do not have a mock community, I am going to show the steps but not actual results will came out of them.

**1. Calling the mock community**

To obtain the  mock community, which is part of the sequencing data we can use:

    mothur> get.groups(count=stability.trim.contigs.good.unique.good.filter.unique.precluster.denovo.vsearch.pick.pick.count_table, fasta=stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta, groups=Mock)

NOTE: for our case of studio the dataset does not contain a mock community, thus this step will generate an empty file (figure 1)

<p align="center">
   <img src="/IMAGES/nt4_1.jpg"><br>
   <b>Figure 1. </b><br>
</p>

Output:
    <ul> 
     <li>stability.trim.contigs.good.unique.good.filter.unique.precluster.denovo.vsearch.pick.pick.pick.count_table</li>
     <li>stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.pick.fasta</li>
    </ul>
  
The *"XXX.count_table"* file tell us how many unique sequences and reads we have.
  
**2. Comparing the unique sequences to what we know of the mock community**

In this step we compare the counts we have in the *".count_table"* with the information we have of the mock community. For instance, when preparing this library we knew how many different strains or species we put into it and at what proportion. Thus in the hypothetic case that our sequencing did not have errors, the count_file should show these known proportions. The HMP_MOCK.V35.FASTA contains the expected data of our community use for comparison.

    mothur > seq.error(fasta=stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.pick.fasta, count=stability.trim.contigs.good.unique.good.filter.unique.precluster.denovo.vsearch.pick.pick.pick.count_table, reference=HMP_MOCK.v35.fasta, aligned=F)

Output:
    <ul>
    <li> stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.pick.error.summary </li>
    <li> stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.pick.error.seq </li>
    <li> stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.pick.error.chimera </li>
    <li> stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.pick.error.seq.forward </li>
    <li> stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.pick.error.seq.reverse </li>
    <li> stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.pick.error.count </li>
    <li> stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.pick.error.matrix </li>
    <li> stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.pick.error.ref </li>
    </ul>

The summary file contains the imformation that is more important with the results of the errors.

**3. Mock Community OTUS information**

To check how many OTUS were detected in the sequenced Mock community,a set of 4 commands have to be run

    mothur> dist.seqs(fasta=stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.pick.fasta, cutoff=0.03)
    
Output 1:

   <ul><li>stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.pick.dist</li></ul>
 
    mothur> cluster(column=stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.pick.dist, count=stability.trim.contigs.good.unique.good.filter.unique.precluster.denovo.vsearch.pick.pick.pick.count_table)
    
Output 2: 

   <ul>
    <li>stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.pick.opti_mcc.list</li>
    <li>stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.pick.opti_mcc.steps</li>
    <li>stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.pick.opti_mcc.sensspec</li>
   </ul>

    mothur> make.shared(list=stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.pick.opti_mcc.list, count=stability.trim.contigs.good.unique.good.filter.unique.precluster.denovo.vsearch.pick.pick.pick.count_table, label=0.03)

Output 3: 
    
   <ul>
   <li>stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.pick.opti_mcc.shared</li>
   </ul>
    
    mothur> rarefaction.single( shared=stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.pick.opti_mcc.shared
    
Output 4:
    
   <ul>
   <li>stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.pick.opti_mcc.groups.rarefaction</li>
   </ul>
    
After running the 4 command to view the results we use:
    
    mothur > system(cat stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.pick.opti_mcc.groups.rarefaction )

The following images shows the results of a mock community and their sequencing results.

<table>
    <td>
        <tr><img align="center" src="/IMAGES/nt4_3.jpg" ></tr>
        <tr><img align="center" src="/IMAGES/nt4_4.jpg" ></tr>
    </td>
</table>
    
The last row shows that from a total of 4048 sequences it was detected 35 OTUS, the physical community was prepared mixing 20 different species of bacteria. According to Shloss this a good result.

Once stimated the error of the sequencing and data handle, we dont need these sequences. Thus, to removed them from our data set we use:

    mothur > remove.groups(count=stability.trim.contigs.good.unique.good.filter.unique.precluster.denovo.vsearch.pick.pick.count_table, fasta=stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta, taxonomy=stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pds.wang.pick.taxonomy, groups=Mock)

<img align="center" src="/IMAGES/nt4_5.jpg"> 

Output:

<ul>
<li> stability.trim.contigs.good.unique.good.filter.unique.precluster.denovo.vsearch.pick.pick.pick.count_table </li>
<li>stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.pick.fasta</li>
<li>stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pds.wang.pick.pick.taxonomy</li>
</ul>

## OTUS picking

In this part we are going to group our sequences into OTUS **But!**: Since our dataset did not have a mock community, the fasta file created is actually empty. In this case I will run the dist.seqs command on my fasta file from the last section of the previous post, as shown in (b).

**1. dist.seq**

a) In case of having a mock community, and after removing them from our dataset use the following command:

    mothur> dist.seqs(fasta=stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.pick.fasta, cutoff=0.03)

Output:
<ul>
    <li> stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.pick.dist</li>
</ul>

b) If our experiment did not hava a mock community ... use:

    mothur> dist.seqs(fasta=stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta, cutoff=0.03)
 
Output:

<ul>
    <li> stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.dist</li>
</ul>

<center>
<table>
    <td>
        <tr><img align="center" src="/IMAGES/nt4_6.jpg"></tr>
        <tr><img align="center" src="/IMAGES/nt4_7.jpg"></tr>
    </td>
</table>
</center>


**2. Cluster**

From now all the command and outputs belongs to a type of pipeline where we didn't have a mock community (note that the extension of the files are xxx.pick.pick.* instead of xxx.pick.pick.pick.* which are regular for a pipeline with mock community)


    mothur> cluster(column=stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.dist, count=stability.trim.contigs.good.unique.good.filter.unique.precluster.denovo.vsearch.pick.pick.count_table)

Output:

<ul>
    <li>stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.opti_mcc.list</li>
    <li>stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.opti_mcc.steps</li>
    <li>stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.opti_mcc.sensspec</li>
</ul>

<Center>
<table>
    <td>
        <tr><img src="/IMAGES/nt4_8.jpg" ></tr>
    </td>
</table>
</center>

**3. Shared**

    mothur> make.shared(list=stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.opti_mcc.list, count=stability.trim.contigs.good.unique.good.filter.unique.precluster.denovo.vsearch.pick.pick.count_table, label=0.03)

Output:
<ul>
    <li>stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.opti_mcc.shared</li>
</ul>

**4. Classify**

    mothur> classify.otu(list=stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.opti_mcc.list, count=stability.trim.contigs.good.unique.good.filter.unique.precluster.denovo.vsearch.pick.pick.count_table, taxonomy=stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pds.wang.pick.taxonomy, label=0.03) 

Output:
<ul>
<li>stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.opti_mcc.0.03.cons.taxonomy</li>
<li>stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.opti_mcc.0.03.cons.tax.summary</li>
</ul>

The following figure shows the content of *"*.tax.summary"* file sorted for the taxon level 3 (Class)

<img align="center" src="/IMAGES/nt4_tax_summary.jpg">



## Creating a phylogenetic tree

Once we have our data classified into OTUS, we can start analyzing it. In order to use distant metrics, we need to create a phylogentic tree. The following steps are one option to create it.

    mothur> dist.seqs(fasta=stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta, output=lt)

Output:
    
<ul>
    <li>stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.phylip.dist</li>
</ul>

    mothur> clearcut(phylip=stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.phylip.dist)

Output: 
<ul>
    <li>stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.phylip.tre</li></ul>

Date: October 02,2019

[:house: Back to home](https://github.com/mhchavez/SMCA-notebook1/wiki)
