# MOTHUR 1: ORGANIZING THE DATA 

**OBJECTIVE:** Understand and organize the dataset to work in MOTHUR

**Before starting:** 
To work in mothur, first we need to obtain our data and define what tool we are using. For the purpose of this first notebook, we are going to use a data provided in class called CHIPMUNK DATA, we already have this fastq files in our working directory (see section C of Cluster Navigation and Secure copy) , and the pipeline will be run in MOTHUR using a stadard procedure based on MiSeq_SOP (https://mothur.org/wiki/MiSeq_SOP) published by Pat Schloss'.

**Let's go!**

## 1) SAMPLE DESCRIPTION

**The chipmunk data:** is from a very small study looking at some chipmunks in the Inland Northwest (USA). It contains fecal microbiome data from two different subspecies of red-tailed chipmunk. This data are the raw reads (forward and reverse) obtaining from sequencing.

In the next table is some information about the animals, the "metadata", including their subspecies (Tamias ruficaudus ruficaudus or T. r. simulans), where the animals were trapped, their mitochondrial haplotype and their sex.

<p align="center"> **Table 1. Metadata for sample data** </p>

<center>
<table>
         <td>Sample</td><td>subspecies</td><td>Trapped</td><td> putmt</td><td>Sex</td>
    <tr> <td>DMNS16474</td><td> simulans</td><td>MoscowMtn</td><td> TRS</td><td>M</td>
    <tr> <td>DMNS16475</td><td> simulans</td><td>MoscowMtn</td><td> TRS</td><td>M</td>
    <tr> <td>DMNS16476</td><td> simulans</td><td>MoscowMtn</td><td> TRS</td><td>M</td>
    <tr> <td>DMNS16477</td><td> simulans</td><td>MoscowMtn</td><td> TRS</td><td>F</td>
    <tr> <td>DMNS16478</td><td> simulans</td><td>MoscowMtn</td><td> TRS</td><td>F</td>
    <tr> <td>DMNS16479</td><td> simulans</td><td>WeitasCrk</td><td> TRR</td><td>F</td>
    <tr> <td>DMNS16482</td><td> ruficaudus</td><td>SlateCrk</td><td> TRR</td><td>M</td>         
    <tr> <td>DMNS16483</td><td> ruficaudus</td><td>SlateCrk</td><td> TRR</td><td>M</td>   
    <tr> <td>DMNS16484</td><td> ruficaudus</td><td>SlateCrk</td><td> TRR</td><td>F</td>
</table>
</center>



To start analyzing data in mothur we need to construct a file called **stability.file** -the name can change to any other that you like- this is the first step before actually working with mothur. bellow there are two methods for creating this file.
But first we need to know what are the names of the reads that we got. This is in the Chipmunk data set moved into our working directories in the cluster.

## 2) CREATING A STABILITY FILE

<p align="center"><img  src="/IMAGES/readschipmunk.PNG"/></p>

The  stability file will organize the data by name of sample and the forward and reverse fastq files associated to that sample.

### First method:

1. Creating a stability file with notepad called *stability.txt*

<p align="center"><img  src="/IMAGES/stabilitytxt.PNG"/></p>

2. Transfer stability file in ~/usr11/chipmunk_fastq using WinSCP
3. To check the content of the transferred file go to: ~home/CAM/mcb5896usr11/chipmunk_fastq/ and type the following command
         
         Bash$ cat.stability.txt

<p align="center"><img width="60%" src="/IMAGES/catstability1.PNG"/></p>

### Second method:

once in the cluster and out of the node, find our directory with the data, then:

1. Load mothur 
         
         Bash$ module load mothur/1.42.3 
         Bash$ mothur 
         mothur> make.file(inputdir=chipmunk_fastq, type=fastq, prefix=stability) 
   
   NOTE: quit from mothur to move into chipmunk_fastq (use cd and ls commands)
   This command can be done jsut if you have the original information named in a fashion that allows mothur to assign the reads with the    samples
   
2. Checking the created file 

        Bash$ cat stability.files
    
 <p align="center"><img align="center" src="/IMAGES/catstability2.PNG"/></p>

[:house: Back to home](https://github.com/mhchavez/SMCA-notebook1/wiki)
