# MOTHUR 3: WORKING WITH CLEANER DATA

**Objective:** 
- Quality control our previously generated dataset

**Recapitulating:** 

In our folder now we have good data that we are going to use for further process located in my working directory ...

## PRE-Aligment Process

**1. Creating list of unique sequences**

Mothur will identified sequences that are duplicated and merge them

        mothur > unique.seqs(fasta=stability.trim.contigs.good.fasta)
        
<p align="center"><img src="/IMAGES/nt3_1.PNG"/> </p>

Output:
          <ul>
                <li>stability.trim.contigs.good.names (which read with which sequence)</li>
                <li>stability.trim.contigs.good.unique.fasta</li>
          </ul>
          
 
**2. Count seqs**
summarizing the data created for better process

        mothur > count.seqs(name=stability.trim.contigs.good.names, group=stability.contigs.good.groups)
        
<img src="/IMAGES/nt3_2.PNG" class="center"/>
 
Output:

          <ul>
                <li>stability.trim.contigs.good.count_table</li>
          </ul>
   
       
to view how the files are  
 
        mothur > system (head stability.trim.contigs.good.count_table)

<p align="center"> <img src="/IMAGES/nt3_3.PNG" class="center"/></p>
 
 **3. Summarizing again**
 
        mothur> summary.seqs(fasta=stability.trim.contigs.good.unique.fasta, count=stability.trim.contigs.good.count_table)
        
        
<p align="center"> <img src="/IMAGES/nt3_5.PNG" class="center"/> </p>
 
Output:
          <ul>
                <li>stability.trim.contigs.good.unique.summary</li>
          </ul>

 
 ## Aligment and quality control
 
 **1. Obtaining reference data**
 
a) Before aligment we need to have a dataset to align with, for the purpose of the class we will use the silva bacteria dataset copied from the class directory                                   
      
          bash-4.2$ cd chipmunk_fastq/
          bash-4.2$ cp /home/CAM/mcb5896/classFiles/Trainset9_032012.pds/trainset9_032012.pds.fasta .
          bash-4.2$ cp /home/CAM/mcb5896/classFiles/Trainset9_032012.pds/trainset9_032012.pds.tax .
          bash-4.2$ cp /home/CAM/mcb5896/classFiles/silva.bacteria/silva.bacteria.fasta .
  
b) Then, since SILVA accomodate more data than we need, we will tell mothur to work just with data for the V4 region
  
          mothur> pcr.seqs(fasta=silva.bacteria.fasta ,start=11894, end=25319,keepdots=F)

<p align="center"> <img src="/IMAGES/nt3_6.PNG" class="center"/> </p>
 
 Output:
          <ul>
                <li>silva.bacteria.pcr.fasta</li>
                <li></li>
          </ul>

 
c) And we will save the file with other name:
 
           mothur> rename.file(input=silva.bacteria.pcr.fasta, new=silva.v4.fasta)
           
d) Checking the new file
           
           mothur> summary.seqs(fasta=silva.v4.fasta)
           
<p align="center"><img src="/IMAGES/nt3_7.PNG" class="center"/></p>


**2. Aligning:**
 
           mothur> align.seqs(fasta=stability.trim.contigs.good.unique.fasta, reference=silva.v4.fasta)
           
Output:
          <ul>
                <li>stability.trim.contigs.good.unique.align</li>
                <li>stability.trim.contigs.good.unique.align.report</li> 
                <li>stability.trim.contigs.good.unique.flip.accnos</li>
          </ul>


**3. Summarizing the aligment**

         mothur> summary.seqs(fasta=stability.trim.contigs.good.unique.align, count=stability.trim.contigs.good.count_table)

<p align="center" ><img src="/IMAGES/nt3_8.PNG" class="center"/></p>


**4. Second Summary and filter**

This is to remove those reads that did not align properly, not ethat maxhomop will look for homopolymers longer than 8.
           
         mothur>  screen.seqs(fasta=stability.trim.contigs.good.unique.align, count=stability.trim.contigs.good.count_table, summary=stability.trim.contigs.good.unique.summary, start=1968, end=11550, maxhomop=8)

<img src="/IMAGES/nt3_9.PNG" class="center"/>

Output:
          <ul>
           <li>stability.trim.contigs.good.unique.good.summary</li> 
           <li>stability.trim.contigs.good.unique.good.align</li> 
           <li>stability.trim.contigs.good.unique.bad.accnos</li> 
                  <li>stability.trim.contigs.good.good.count_tablen</li> 
          </ul>
  
 **Filtering**
 
        mothur>filter.seqs(fasta=stability.trim.contigs.good.unique.good.align, vertical=T, trump=.)
 
 <p align="center" ><img src="/IMAGES/nt3_10.PNG" class="center"/></p>
 
Output:
          <ul>
           <li>stability.filter</li> 
           <li>stability.trim.contigs.good.unique.good.filter.fasta</li> 
           </ul>
       
        
**5. Removing redundant seqs**

        mothur> unique.seqs(fasta=stability.trim.contigs.good.unique.good.filter.fasta,count=stability.trim.contigs.good.good.count_table)

<p align="center" ><img src="/IMAGES/n3_11.PNG" class="center"/></p>

Note in the figure that 72 sequences were redundant

 Ouput:
          <ul>
           <li>stability.trim.contigs.good.unique.good.filter.count_table</li>
           <li>stability.trim.contigs.good.unique.good.filter.unique.fasta</li> 
           </ul>
   

**6. Pre-clustering**

It is recommended to allow 1 base different every 100, since our reads are ~250 of lenght,Therefore we set our data with 2 differences. All reads that are only 2 base different will be cluster as one.

        mothur> pre.cluster(fasta=stability.trim.contigs.good.unique.good.filter.unique.fasta, count=stability.trim.contigs.good.unique.good.filter.count_table, diffs=2)
        
 Output:
          <ul>
        <li><li>stability.trim.contigs.good.unique.good.filter.unique.precluster.fasta</li>
        <li>stability.trim.contigs.good.unique.good.filter.unique.precluster.count_table</li>
        <li>stability.trim.contigs.good.unique.good.filter.unique.precluster.DMNS16474.map</li>
        <li>stability.trim.contigs.good.unique.good.filter.unique.precluster.DMNS16475.map</li>
        <li>stability.trim.contigs.good.unique.good.filter.unique.precluster.DMNS16476.map</li>
        <li>stability.trim.contigs.good.unique.good.filter.unique.precluster.DMNS16477.map</li>
        <li>stability.trim.contigs.good.unique.good.filter.unique.precluster.DMNS16478.map</li>
        <li>stability.trim.contigs.good.unique.good.filter.unique.precluster.DMNS16479.map</li>
        <li>stability.trim.contigs.good.unique.good.filter.unique.precluster.DMNS16482.map</li>
        <li>stability.trim.contigs.good.unique.good.filter.unique.precluster.DMNS16483.map</li>
        <li>stability.trim.contigs.good.unique.good.filter.unique.precluster.DMNS16484.map</li> 
           </ul>
    
 Other important results:
 
        It took 0 secs to merge 72270 sequences group data.
        /******************************************/
        Running command: get.seqs(fasta=stability.trim.contigs.good.unique.good.filter.unique.fasta,         accnos=stability.trim.contigs.good.unique.good.filter.unique.precluster.count_table.temp)       
        Selected 64419 sequences from your fasta file.

        Output File Names:
        stability.trim.contigs.good.unique.good.filter.unique.pick.fasta

        /******************************************/
        It took 203 secs to run pre.cluster.

**7. Removing Chimeras**

we will localize and remove the chimeras from the count file with:

        mothur> chimera.vsearch(fasta=stability.trim.contigs.good.unique.good.filter.unique.precluster.fasta, count=stability.trim.contigs.good.unique.good.filter.unique.precluster.count_table, dereplicate=t)

OUTPUT: 
<ul>
        <li> stability.trim.contigs.good.unique.good.filter.unique.precluster.denovo.vsearch.pick.count_table </li>
        <li>stability.trim.contigs.good.unique.good.filter.unique.precluster.denovo.vsearch.chimeras</li>
        <li>stability.trim.contigs.good.unique.good.filter.unique.precluster.denovo.vsearch.accnos</li>
</ul>


then to actually remove the localized chimeras from the fasta file we run:

        mothur> remove.seqs(fasta=stability.trim.contigs.good.unique.good.filter.unique.precluster.fasta, accnos=stability.trim.contigs.good.unique.good.filter.unique.precluster.denovo.vsearch.accnos)

<p align="center" > <img src="/IMAGES/nt3_12.PNG" class="center"/></p>
 
OUTPUT:

<ul>
        <li> stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.fasta</li>
</ul>

and now the result:

        mothur> summary.seqs(fasta=stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.fasta,count=stability.trim.contigs.good.unique.good.filter.unique.precluster.denovo.vsearch.pick.count_table)

<img src="/IMAGES/n3_13.PNG" class="center"/>
 
 OUTPUT:

<ul>
        <li> stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.summary</li>
</ul>

**8. Classifying and removing data**

Note: make sure before running the command that the files Trainset ** are in the Folder
Here we are classifying our sequences into taxons at a 80% of certaintity, using as reference the "Trainset database"
        
        mothur> classify.seqs(fasta=stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.fasta, count=stability.trim.contigs.good.unique.good.filter.unique.precluster.denovo.vsearch.pick.count_table, reference= trainset9_032012.pds.fasta, taxonomy= trainset9_032012.pds.tax, cutoff=80)
  
<p align="center"> <img src="/IMAGES/n3_14.jpg" class="center"/></p>
  
OUTPUT:
<ul>
        <li>stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pds.wang.taxonomy</li>
        <li>stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pds.wang.tax.summary</li>
</ul>

Now we will remove the squences that might be chloroplast, mitochondria and other unwanted.

        mothur > remove.lineage(fasta=stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.fasta , count=stability.trim.contigs.good.unique.good.filter.unique.precluster.denovo.vsearch.pick.count_table ,taxonomy=stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pds.wang.taxonomy ,taxon=Chloroplast-Mitochondria-unknown-Archaea-Eukaryota)
        
<p align="center" ><img src="/IMAGES/n3_15.jpg" class="center"/></p>
  
OUTPUT:
<ul>
        <li>stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pds.wang.pick.taxonomy</li>
        <li>stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pds.wang.accnos</li>
        <li>stability.trim.contigs.good.unique.good.filter.unique.precluster.denovo.vsearch.pick.pick.count_table</li>
        <li>stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta</li>
</ul>

**9 Taxonomic File!**

        mothur> summary.tax(taxonomy=current, count=current)
 
 <p align="center"><img src="/IMAGES/n3_16.jpg" class="center"/></p>
        
Output:
<ul>
    <li>       stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pds.wang.pick.tax.summary</li>
</ul>

Download the file and check it on Excel

<p align="center"> <img src="/IMAGES/n3_17.jpg" class="center"/></p>
